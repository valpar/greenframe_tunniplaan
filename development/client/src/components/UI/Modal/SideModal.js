import { useState } from "react";
import { Fragment, useEffect } from "react";
import ReactDOM from "react-dom";

const Backdrop = (props) => {
  useEffect(() => {
    const themeColorMeta = document.querySelector('meta[name="theme-color"]');

    themeColorMeta.setAttribute("content", "#5C5C5C");

    return () => {
      themeColorMeta.setAttribute("content", "#fff");
    };
  }, []);

  return (
    <div
      className="fixed top-0 left-0 w-full h-screen z-20 bg-neutral-900 opacity-70"
      onClick={props.onClose}
    />
  );
};

const ModalOverlay = (props) => {
  return (
    <div
      data-testid="modal-overlay"
      className={`${
        props.onHidden ? "animate-slideRight" : "animate-slideRightOff"
      } fixed top-20 w-full h-full left-1/2 -translate-x-1/2 bg-white z-30 shadow-lg`}
    >
      <div
        className={`absolute ${
          props.modalOverlay ? "flex" : "hidden"
        } justify-center items-center w-full h-full bg-neutral-900 opacity-70 z-30`}
      ></div>
      <div className="p-4">
        <div>{props.children}</div>
      </div>
    </div>
  );
};
const portalElement = document.getElementById("overlays");

const SideModal = (props) => {
  return (
    <Fragment>
      {ReactDOM.createPortal(
        <ModalOverlay onHidden={props.onHidden} modalOverlay={props.overlay}>
          {props.children}
        </ModalOverlay>,
        portalElement
      )}
    </Fragment>
  );
};

export default SideModal;
