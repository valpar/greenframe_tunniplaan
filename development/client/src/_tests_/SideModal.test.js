import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import SideModal from '../components/UI/Modal/SideModal';

beforeAll(() => {
    const meta = document.createElement('meta');
    meta.setAttribute('name', 'theme-color');
    document.head.appendChild(meta);
  });
  
afterAll(() => {
  // eslint-disable-next-line testing-library/no-node-access
  document.head.removeChild(document.querySelector('meta[name="theme-color"]'));
});

// kui seda ei tee (createPortal), siis target "is not DOM element"
jest.mock('react-dom', () => ({
  ...jest.requireActual('react-dom'),
  createPortal: (element, target) => element, 
}));

describe('Modal', () => {
  it("matches snapshot", () => {
    const tree = renderer.create(<SideModal />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders with Backdrop and ModalOverlay', () => {
    render(<SideModal />);
    expect(screen.getByTestId('modal-overlay')).toBeInTheDocument();
  });
  it('renders with different values for onHidden and modalOverlay props', () => {
    // Render with onHidden=true and modalOverlay=false
    render(<SideModal onHidden={true} modalOverlay={false} />);
    const modalOverlay = screen.getByTestId('modal-overlay');

    // sellise klassi saab komponent
    expect(modalOverlay).toHaveClass('animate-slideRight fixed top-20 w-full h-full left-1/2 -translate-x-1/2 bg-white z-30 shadow-lg');
    expect(modalOverlay).not.toHaveClass('flex');
  });

  // lisada juurde
});