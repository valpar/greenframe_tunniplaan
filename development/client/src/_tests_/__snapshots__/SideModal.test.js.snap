// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal matches snapshot 1`] = `
<div
  className="animate-slideRightOff fixed top-20 w-full h-full left-1/2 -translate-x-1/2 bg-white z-30 shadow-lg"
  data-testid="modal-overlay"
>
  <div
    className="absolute hidden justify-center items-center w-full h-full bg-neutral-900 opacity-70 z-30"
  />
  <div
    className="p-4"
  >
    <div />
  </div>
</div>
`;
